flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -142,4 +143,16 @@
      */
     String getSelectFromStatement(
             String tableName, String[] selectFields, String[] conditionFields);
+
+    /** Create catalog instance. */
+    default AbstractJdbcCatalog createCatalog(
```

Comment:
Logically this seems misplaced to me, because creating a catalog doesn't relate to the definition of a dialect. However, I don't have a better suggestion either, we shouldn't introduce even more custom infrastructure.

Reply:
thx for your review... 
yes, u are right. If we add new interface, we need to modify a lot code. I think the current way might be better than modifying a lot of code.